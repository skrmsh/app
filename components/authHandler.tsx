import axios, { AxiosError, AxiosResponse } from 'axios';
import React, { useState } from 'react';
import { StyleSheet } from 'react-native';
import {
  ActivityIndicator,
  Avatar,
  Button,
  Text,
  TextInput,
} from 'react-native-paper';
import { ErrorDialog } from './';

type AuthHandlerProps = {
  authToken: string;
  setAuthToken: (e: string) => void;
  saveAuthToken: (e: string) => void
};

function authenticate(
  username: string,
  password: string,
  errorCallback: (e: AxiosError) => void,
  successCallback: (e: void | AxiosResponse) => void,
) {
  console.log('logging in...');
  axios
    .post('https://olel.de/auth', { email: username, password: password })
    .catch(errorCallback)
    .then(successCallback);
}

export const AuthHandler = ({ authToken, setAuthToken, saveAuthToken}: AuthHandlerProps) => {
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');
  const [authLoading, setAuthLoading] = useState(false);
  const [showingError, setShowingError] = useState(false);
  const [errorMsg, setErrorMsg] = useState('');
  const [playerName, setPlayerName] = useState('');

  function requestPlayerInfo(accessToken) {
    const config = {
      headers: {
        'x-access-token': accessToken,
      },
    };
    console.log(config);
    axios.get('https://olel.de/user', config).then((e: AxiosResponse) => {
      console.log(e.data);
      setPlayerName(e.data.username);
    });
  }
  if (authToken) {
    requestPlayerInfo(authToken);
  }

  return (
    <>
      <ErrorDialog
        showingError={showingError}
        setShowingError={setShowingError}
        errorMsg={errorMsg}
      />
      {!!playerName ? (
        <>
          <Text variant="displayMedium">Hi, {playerName}!</Text>
          <Avatar.Image
            size={150}
            source={require('../utils/avatars/default02.png')}
          />
          <Button
            mode="contained"
            onPress={() => {
              setAuthToken('');
              setPlayerName('');
              saveAuthToken('')
            }}
            disabled={!authToken}>
            Logout
          </Button>
        </>
      ) : (
        <>
          <TextInput
            style={styles.input}
            onChangeText={setUsername}
            value={username}
            mode="outlined"
          />
          <TextInput
            style={styles.input}
            onChangeText={setPassword}
            value={password}
            mode="outlined"
            secureTextEntry
          />
          <Button
            mode="contained"
            onPress={() => {
              setAuthLoading(true);
              authenticate(
                username,
                password,
                (e: AxiosError) => {
                  console.log('http error:', e.code, e.message);
                  setErrorMsg(`http error:, ${e.code}, ${e.message}`);
                  setShowingError(true);
                  setAuthLoading(false);
                },
                (e: void | AxiosResponse) => {
                  if (e) {
                    setAuthToken(e.data.access_token);
                    console.log('Successfully authenticated!');
                    setAuthLoading(false);
                    requestPlayerInfo(e.data.access_token);
                    saveAuthToken(e.data.access_token)
                  }
                },
              );
            }}
            disabled={!!authToken}>
            Login
          </Button>
        </>
      )}
      {authLoading ? <ActivityIndicator size="large" /> : <></>}
    </>
  );
};
const styles = StyleSheet.create({
  input: {
    height: 40,
    marginBottom: 8,
    padding: 12,
  },
  button: {
    margin: 12,
  },
});
