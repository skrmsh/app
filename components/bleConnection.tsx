import React, { useEffect, useState } from 'react';
import {
  Button,
  Card,
  Modal,
  Portal,
  Text,
  useTheme,
} from 'react-native-paper';
import { getStyles } from '../utils';
import { LoadingDialog } from './loadingDialog';
import SKBLEManager, { SKBLEDev } from '../utils/bleManager';

type BleConnectionProps = {
  connectionIsFor: string;
};

export const BleConnection = ({
  connectionIsFor,
}: BleConnectionProps): JSX.Element => {
  const [connectionPopupVisible, setConnectionPopupVisible] = useState(false);

  const [currentlyLoading, setCurrentlyLoading] = useState(false);

  const [selectedDevice, setSelectedDevice] = useState<SKBLEDev>();
  const [connectionState, setConnectionState] = useState<boolean>(false);
  const [discoveredDevices, setDiscoveredDevices] = useState<Array<SKBLEDev>>(
    [],
  );

  const theme = useTheme();

  useEffect(() => {
    SKBLEManager.Instance.onDeviceDiscovery(device => {
      setDiscoveredDevices([...SKBLEManager.Instance.discoveredDevices]);
    });
  }, []);

  useEffect(() => {
    if (!!!selectedDevice) return;
    SKBLEManager.Instance.onDeviceConnected(device => {
      console.log(
        `bleConnection was informed that ${device.name} was connected`,
      );
      if (device.id === selectedDevice.id) {
        setConnectionState(true);
      }
    });
    SKBLEManager.Instance.onDeviceDisconnected(device => {
      console.log(
        `bleConnection was informed that ${device.name} was disconnected`,
      );
      if (device.id === selectedDevice.id) {
        setConnectionState(false);
      }
    });
  }, [selectedDevice]);

  useEffect(() => {
    console.log('currently known devices: ', discoveredDevices);
  }, [discoveredDevices]);

  useEffect(() => {
    if (connectionPopupVisible) {
      setSelectedDevice(undefined);
      SKBLEManager.Instance.startScan();
    } else {
      SKBLEManager.Instance.stopScan();
    }
  }, [connectionPopupVisible]);

  return (
    <>
      <Portal>
        <LoadingDialog showingLoading={currentlyLoading} />
        <Modal
          visible={connectionPopupVisible}
          onDismiss={() => setConnectionPopupVisible(false)}
          contentContainerStyle={getStyles(theme).modalContainer}>
          <Text style={{ marginBottom: 20 }}>Select your Device</Text>
          {discoveredDevices.map(device => (
            <Button
              key={device.id}
              icon="pistol"
              mode={selectedDevice?.id === device.id ? 'contained' : 'outlined'}
              onPress={() => {
                setSelectedDevice(
                  selectedDevice?.id === device.id ? undefined : device,
                );
                setConnectionState(false);
              }}>
              {device.name}
            </Button>
          ))}

          <Button
            disabled={!selectedDevice}
            onPress={() => {
              if (selectedDevice) {
                setCurrentlyLoading(true);
                SKBLEManager.Instance.connectToDiscoveredDevice(selectedDevice);
                setConnectionPopupVisible(false);
                setCurrentlyLoading(false);
              }
            }}>
            Connect
          </Button>
          <Button
            onPress={() => {
              setConnectionPopupVisible(false);
            }}>
            Cancel
          </Button>
        </Modal>
      </Portal>

      <Card style={getStyles(theme).connectionCard}>
        <Card.Title title={`${connectionIsFor} Connection`} />
        <Card.Content>
          {connectionState ? (
            <Text>Connected to {selectedDevice?.name}</Text>
          ) : (
            <Text>Not Connected!</Text>
          )}
        </Card.Content>
        <Card.Actions>
          {connectionState ? (
            <Button
              onPress={() => {
                SKBLEManager.Instance.disconnectFromDevice(selectedDevice!);
              }}>
              Disconnect
            </Button>
          ) : (
            <Button onPress={() => setConnectionPopupVisible(true)}>
              Connect to {connectionIsFor}
            </Button>
          )}
        </Card.Actions>
      </Card>
    </>
  );
};
