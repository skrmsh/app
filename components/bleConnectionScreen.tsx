import React, { useEffect, useState } from 'react';
import { View } from 'react-native';
import { BleManager, Device } from 'react-native-ble-plx';
import { Button } from 'react-native-paper';
import { BleConnection, ErrorDialog } from '.';
import { getStyles, killManager, startBluetooth } from '../utils';
import { Theme } from '@react-navigation/native';

type BleConnectionScreenProps = {
  manager: BleManager | undefined;
  setManager: (manager: BleManager | undefined) => void;
  messageCallback: (message: string) => void;
  connectedDevices: Device[];
  setConnectedDevices: (setter: (devices: Device[]) => Device[]) => void;
  onScreenFinishedCallback: () => void;
  theme: Theme;
};
export const BleConnectionScreen = ({
  manager,
  setManager,
  messageCallback,
  connectedDevices,
  setConnectedDevices,
  onScreenFinishedCallback,
  theme,
}: BleConnectionScreenProps): JSX.Element => {
  console.log(connectedDevices);
  const [errorMessage, setErrorMessage] = useState('');
  useEffect(() => {
    if (!manager) {
      startBluetooth(setManager);
    }
  }, []);
  return (
    <>
      <View
        style={{
          margin: 15,
          height: '100%',
          ...getStyles(theme).centerContainer,
        }}>
        <ErrorDialog
          showingError={!!errorMessage}
          setShowingError={() => setErrorMessage('')}
          errorMsg={errorMessage}
        />
        <BleConnection
          connectionIsFor="Phaser"
          onConnectCallback={device => {}}
          onDisconnectCallback={device => {}}
          manager={manager}
          messageCallback={messageCallback}
          errorMessageSetter={setErrorMessage}
          connectedDevices={connectedDevices}
          setConnectedDevices={setConnectedDevices}
          theme={theme}
        />
        <BleConnection
          connectionIsFor="Vest"
          onConnectCallback={device => {}}
          onDisconnectCallback={device => {}}
          manager={manager}
          messageCallback={messageCallback}
          errorMessageSetter={setErrorMessage}
          connectedDevices={connectedDevices}
          setConnectedDevices={setConnectedDevices}
          theme={theme}
        />
        <Button
          onPress={onScreenFinishedCallback}
          mode="contained"
          textColor={theme.colors.text}
          theme={theme}
          style={{
            marginRight: 40,
            marginTop: 10,
            alignSelf: 'flex-end',
          }}
          disabled={connectedDevices.length < 1}>
          Continue
        </Button>
      </View>
    </>
  );
};
