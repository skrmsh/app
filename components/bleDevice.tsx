import React from 'react';
import { Device } from 'react-native-ble-plx';
import { Button, Card, Text, useTheme } from 'react-native-paper';
import { getStyles } from '../utils';
type BleDeviceProps = {
  device: Device;
  isConnected: boolean;
  connect: (d: Device) => void;
  disconnect: (d: Device) => void;
};
export const BleDevice = ({
  device,
  isConnected,
  connect,
  disconnect,
}: BleDeviceProps) => {
  const theme = useTheme();
  return (
    <>
      <Card key={device.id} style={getStyles(theme).cardcontent}>
        <Card.Content>
          <Text variant="titleLarge" key={'deviceid'}>
            Device ID: {device.id}
          </Text>
          <Text variant="bodyMedium" key={'devicename'}>
            {device.name ? device.name : 'no name supplied'}
          </Text>
          <Text variant="bodyMedium" key={'connectionstatus'}>
            {isConnected ? 'Connected' : 'Not Connected'}
          </Text>
          <Card.Actions key={'connect'}>
            <Button
              mode="contained"
              key={'bruteforceconnect'}
              onPress={() => connect(device)}
              disabled={isConnected}
              testID={'connectButton'}>
              Connect
            </Button>
          </Card.Actions>
          <Card.Actions key={'disconnect'}>
            <Button
              key={'disconnect'}
              onPress={() => disconnect(device)}
              disabled={!isConnected}
              testID={'disconnectButton'}>
              Disconnect
            </Button>
          </Card.Actions>
        </Card.Content>
      </Card>
    </>
  );
};
