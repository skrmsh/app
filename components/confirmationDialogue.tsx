import React, { useState } from "react"
import { Button, Dialog, Portal, Text, useTheme } from "react-native-paper"

type ConfirmationDialogueProps = {
    showing: boolean,
    setShowing: (e: boolean) => void,
    confirmationText: string,
    action: () => void
}

export const ConfirmationDialogue = ({ showing, setShowing, confirmationText, action }: ConfirmationDialogueProps) => {
    const theme = useTheme();
    return (<>
        <Portal>
            <Dialog visible={showing} onDismiss={() => setShowing(false)}>
                <Dialog.Icon
                    icon="alert-octagon"
                    size={32}
                    color={theme.colors.error}
                />
                <Dialog.Content>
                    <Text variant="bodyMedium">{confirmationText}</Text>
                </Dialog.Content>
                <Dialog.Actions>
                    <Button key={'closeButton'} onPress={() => setShowing(false)}>
                        Back
                    </Button>
                    <Button key={'closeButton'} onPress={() => { setShowing(false); action() }}>
                        Continue
                    </Button>
                </Dialog.Actions>
            </Dialog>
        </Portal>
    </>)
}