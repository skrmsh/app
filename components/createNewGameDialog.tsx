import axios from 'axios';
import { useState } from 'react';
import {
  Button,
  Dialog,
  Portal,
  Text,
  TextInput,
  useTheme,
} from 'react-native-paper';

import { Picker } from '@react-native-picker/picker';
import { Theme } from '@react-navigation/native';
import { getStyles } from '../utils';

type createNewGameDialogProps = {
  showing: boolean;
  setShowing: (e: boolean) => void;
  possibleGamemodes: string[];
  gamemode: string;
  setGamemode: (e: string) => void;
  callback: (e: string) => void;
  theme: Theme
};

export const CreateNewGameDialog = ({
  showing,
  setShowing,
  callback,
  gamemode,
  setGamemode,
  possibleGamemodes,
  theme
}: createNewGameDialogProps) => {
  return (
    <>
      <Portal>
        <Dialog style={getStyles(theme).dialog} visible={showing} onDismiss={() => setShowing(false)}>
          <Dialog.Icon icon="pencil" size={32} color={theme.colors.primary} />
          <Dialog.Content>
            <Text variant="bodyLarge">Please choose a game type:</Text>
            <Picker
              selectedValue={gamemode}
              onValueChange={(itemValue, itemIndex) => setGamemode(itemValue)}>
              {possibleGamemodes.map((e: string) => (
                <Picker.Item color={theme.colors.text} key={e} label={e} value={e} />
              ))}
            </Picker>
          </Dialog.Content>
          <Dialog.Actions>
            <Button
              key={'saveButton'}
              onPress={() => {
                setShowing(false);
              }}>
              Cancel
            </Button>
            <Button
              key={'createButton'}
              theme={theme}
              mode="contained"
              onPress={() => {
                setShowing(false);
                callback(gamemode);
              }}>
              Create Game
            </Button>
          </Dialog.Actions>
        </Dialog>
      </Portal>
    </>
  );
};
