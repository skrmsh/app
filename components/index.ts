export * from './LoginScreen';
export * from './authHandler';
export * from './bleConnection';
export * from './bleConnectionScreen';
export * from './confirmationDialogue';
export * from './createNewGameDialog';
export * from './errorDialog';
export * from './gidInputBox';
export * from './gameManager';
export * from './globalSettings';
export * from './loadingDialog';
export * from './seperator';
export * from './taskStatusBar';
export * from './webSocketHandler';
