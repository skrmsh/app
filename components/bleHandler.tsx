import React, { useState } from 'react';
import { ActivityIndicator, Button, Text } from 'react-native-paper';

import { sendTimestamp, turnOffAllDevices } from '../utils';
import { BleDevice } from './bleDevice';
import { ErrorDialog } from './errorDialog';
import { Separator } from './seperator';
import SKBLEManager, { SKBLEDev } from '../utils/bleManager';

type BleHandlerProps = {
  setBleEnabled: (e: boolean) => void;
  bleEnabled: boolean;
  messageCallback: (e: string) => void;
  onCompletedCallback: () => void;
};

export const BleHandler = ({
  setBleEnabled,
  bleEnabled,
  messageCallback,
  onCompletedCallback,
}: BleHandlerProps) => {
  const [bleIsLoading, setBleIsLoading] = useState(false);
  const [errorMsg, setErrorMsg] = useState('');
  useState(() => {
    SKBLEManager.Instance.start();
    setBleEnabled(true);
  });

  return (
    <>
      <ErrorDialog
        showingError={!!errorMsg}
        setShowingError={(e: boolean) => {
          if (!e) {
            setErrorMsg('');
          }
        }}
        errorMsg={errorMsg}
      />
      <Separator />
      <Button
        mode="contained"
        onPress={() => {
          SKBLEManager.Instance.startScan();
        }}
        disabled={!bleEnabled}>
        Scan for Phasors
      </Button>
      <Separator />
      <Button
        onPress={() => {
          SKBLEManager.Instance.stopScan();
        }}
        disabled={!bleEnabled}
        mode="contained">
        Stop Scanning
      </Button>
      <Separator />
      <Button
        mode="contained"
        onPress={() => {
          setBleEnabled(false);
          SKBLEManager.Instance.stop();
        }}>
        Kill BLE
      </Button>
      <Text style={{ margin: 15 }}>
        Discovered Devices: {SKBLEManager.Instance.discoveredDevices.length}
      </Text>
      {SKBLEManager.Instance.discoveredDevices.length > 0 ? (
        <>
          <Separator />
          <Button
            mode="contained"
            onPress={() => {
              turnOffAllDevices();
            }}>
            Turn off all Devices
          </Button>
        </>
      ) : (
        <></>
      )}
      {bleIsLoading ? <ActivityIndicator size="large" /> : <></>}
      {SKBLEManager.Instance.discoveredDevices.map(device => (
        <>
          {device.name?.includes(
            'skrm',
          ) /* duplicate, checked by skbleman */ ? (
            <>
              <BleDevice
                device={device}
                isConnected={device.connectionState}
                connect={(d: SKBLEDev) => {
                  setBleIsLoading(true);
                  SKBLEManager.Instance.connectToDiscoveredDevice(d);
                  /*
                  connectUntilSuccess(
                    d,
                    addPhasorToConnectedDevices,
                    manager,
                    () => {
                      setTimeout(() => sendTimestamp(device), 1500);
                      addOnDisconnectCallback(manager, device, (e: Device) => {
                        setErrorMsg(
                          `Lost Connection to Phasor with ID: ${e.id}`,
                        );
                        removePhasorFromConnectedDevices(e);
                      });
                      setBleIsLoading(false);
                    },
                    messageCallback,
                  );
                  */
                }}
                disconnect={(d: SKBLEDev) => {
                  SKBLEManager.Instance.disconnectFromDevice(d);
                }}
              />
              <Separator />
            </>
          ) : (
            <></>
          )}
        </>
      ))}
      {SKBLEManager.Instance.connectedDevices.length > 0 && (
        <Button onPress={() => onCompletedCallback()}>Continue</Button>
      )}
    </>
  );
};
