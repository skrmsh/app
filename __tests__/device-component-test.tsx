/**
 * @format
 */

import React from 'react';
// Note: test renderer must be required after react-native.
import 'react-native';
import {
  BleError,
  Characteristic,
  ConnectionOptions,
  ConnectionPriority,
  Descriptor,
  Device,
  Service,
  Subscription,
} from 'react-native-ble-plx';
import renderer from 'react-test-renderer';
import { BleDevice } from '../components/bleDevice';

export const getDummyDevice = ({
  id = 'testMac',
  name = 'TestName',
}: Partial<Device> = {}): Device => {
  return {
    id: id,
    name: name,
    rssi: null,
    mtu: 0,
    manufacturerData: null,
    serviceData: null,
    serviceUUIDs: null,
    localName: null,
    txPowerLevel: null,
    solicitedServiceUUIDs: null,
    isConnectable: null,
    overflowServiceUUIDs: null,
    requestConnectionPriority: function (
      connectionPriority: ConnectionPriority,
      transactionId?: string | undefined,
    ): Promise<Device> {
      throw new Error('Function not implemented.');
    },
    readRSSI: function (transactionId?: string | undefined): Promise<Device> {
      throw new Error('Function not implemented.');
    },
    requestMTU: function (
      mtu: number,
      transactionId?: string | undefined,
    ): Promise<Device> {
      throw new Error('Function not implemented.');
    },
    connect: function (
      options?: ConnectionOptions | undefined,
    ): Promise<Device> {
      throw new Error('Function not implemented.');
    },
    cancelConnection: function (): Promise<Device> {
      throw new Error('Function not implemented.');
    },
    isConnected: function (): Promise<boolean> {
      throw new Error('Function not implemented.');
    },
    onDisconnected: function (
      listener: (error: BleError | null, device: Device) => void,
    ): Subscription {
      throw new Error('Function not implemented.');
    },
    discoverAllServicesAndCharacteristics: function (
      transactionId?: string | undefined,
    ): Promise<Device> {
      throw new Error('Function not implemented.');
    },
    services: function (): Promise<Service[]> {
      throw new Error('Function not implemented.');
    },
    characteristicsForService: function (
      serviceUUID: string,
    ): Promise<Characteristic[]> {
      throw new Error('Function not implemented.');
    },
    descriptorsForService: function (
      serviceUUID: string,
      characteristicUUID: string,
    ): Promise<Descriptor[]> {
      throw new Error('Function not implemented.');
    },
    readCharacteristicForService: function (
      serviceUUID: string,
      characteristicUUID: string,
      transactionId?: string | undefined,
    ): Promise<Characteristic> {
      throw new Error('Function not implemented.');
    },
    writeCharacteristicWithResponseForService: function (
      serviceUUID: string,
      characteristicUUID: string,
      valueBase64: string,
      transactionId?: string | undefined,
    ): Promise<Characteristic> {
      throw new Error('Function not implemented.');
    },
    writeCharacteristicWithoutResponseForService: function (
      serviceUUID: string,
      characteristicUUID: string,
      valueBase64: string,
      transactionId?: string | undefined,
    ): Promise<Characteristic> {
      throw new Error('Function not implemented.');
    },
    monitorCharacteristicForService: function (
      serviceUUID: string,
      characteristicUUID: string,
      listener: (
        error: BleError | null,
        characteristic: Characteristic | null,
      ) => void,
      transactionId?: string | undefined,
    ): Subscription {
      throw new Error('Function not implemented.');
    },
    readDescriptorForService: function (
      serviceUUID: string,
      characteristicUUID: string,
      descriptorUUID: string,
      transactionId?: string | undefined,
    ): Promise<Descriptor> {
      throw new Error('Function not implemented.');
    },
    writeDescriptorForService: function (
      serviceUUID: string,
      characteristicUUID: string,
      descriptorUUID: string,
      valueBase64: string,
      transactionId?: string | undefined,
    ): Promise<Descriptor> {
      throw new Error('Function not implemented.');
    },
  };
};

const defaultProps = {
  isConnected: true,
  device: getDummyDevice(),
  connect: () => {},
  disconnect: () => {},
};

it('renders correctly and matches snapshot', () => {
  const tree = renderer.create(<BleDevice {...defaultProps} />);
  expect(tree).toMatchSnapshot();
});
