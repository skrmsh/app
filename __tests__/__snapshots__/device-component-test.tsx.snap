// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly and matches snapshot 1`] = `
<View
  collapsable={false}
  style={
    {
      "alignSelf": undefined,
      "bottom": undefined,
      "end": undefined,
      "left": undefined,
      "position": undefined,
      "right": undefined,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 3,
      "start": undefined,
      "top": undefined,
    }
  }
>
  <View
    collapsable={false}
    style={
      {
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "rgba(247, 243, 249, 1)",
          "borderRadius": 12,
          "elevation": 1,
          "marginBottom": 10,
          "marginTop": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": true,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "flexShrink": 1,
          }
        }
        testID="card"
      >
        <View
          style={
            [
              {
                "paddingHorizontal": 16,
              },
              {
                "paddingVertical": 16,
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontSize": 22,
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 28,
                },
                {
                  "textAlign": "left",
                },
                {
                  "writingDirection": "ltr",
                },
                undefined,
              ]
            }
          >
            Device ID: 
            testMac
          </Text>
          <Text
            style={
              [
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "400",
                  "letterSpacing": 0.25,
                  "lineHeight": 20,
                },
                {
                  "textAlign": "left",
                },
                {
                  "writingDirection": "ltr",
                },
                undefined,
              ]
            }
          >
            TestName
          </Text>
          <Text
            style={
              [
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "400",
                  "letterSpacing": 0.25,
                  "lineHeight": 20,
                },
                {
                  "textAlign": "left",
                },
                {
                  "writingDirection": "ltr",
                },
                undefined,
              ]
            }
          >
            Connected
          </Text>
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "padding": 8,
                },
                undefined,
                {
                  "justifyContent": "flex-end",
                },
              ]
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "alignSelf": undefined,
                  "bottom": undefined,
                  "end": undefined,
                  "left": undefined,
                  "position": undefined,
                  "right": undefined,
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                  "start": undefined,
                  "top": undefined,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "shadowColor": "#000",
                    "shadowOffset": {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "backgroundColor": "rgba(28, 27, 31, 0.12)",
                      "borderColor": "transparent",
                      "borderRadius": 20,
                      "borderStyle": "solid",
                      "borderWidth": 0,
                      "marginLeft": 8,
                      "minWidth": 64,
                    }
                  }
                >
                  <View
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": true,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      [
                        {
                          "overflow": "hidden",
                        },
                        false,
                        {
                          "borderRadius": 20,
                        },
                      ]
                    }
                    testID="connectButton"
                  >
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "flexDirection": "row",
                            "justifyContent": "center",
                          },
                          undefined,
                        ]
                      }
                    >
                      <Text
                        numberOfLines={1}
                        selectable={false}
                        style={
                          [
                            {
                              "color": "rgba(28, 27, 31, 1)",
                              "fontFamily": "System",
                              "fontSize": 14,
                              "fontWeight": "500",
                              "letterSpacing": 0.1,
                              "lineHeight": 20,
                            },
                            {
                              "textAlign": "left",
                            },
                            {
                              "writingDirection": "ltr",
                            },
                            [
                              {
                                "marginHorizontal": 16,
                                "marginVertical": 9,
                                "textAlign": "center",
                              },
                              false,
                              {
                                "marginHorizontal": 24,
                                "marginVertical": 10,
                              },
                              false,
                              false,
                              {
                                "color": "rgba(28, 27, 31, 0.38)",
                                "fontFamily": "System",
                                "fontSize": 14,
                                "fontWeight": "500",
                                "letterSpacing": 0.1,
                                "lineHeight": 20,
                              },
                              undefined,
                            ],
                          ]
                        }
                      >
                        Connect
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "padding": 8,
                },
                undefined,
                {
                  "justifyContent": "flex-end",
                },
              ]
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "alignSelf": undefined,
                  "bottom": undefined,
                  "end": undefined,
                  "left": undefined,
                  "position": undefined,
                  "right": undefined,
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                  "start": undefined,
                  "top": undefined,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "shadowColor": "#000",
                    "shadowOffset": {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "backgroundColor": "transparent",
                      "borderColor": "rgba(121, 116, 126, 1)",
                      "borderRadius": 20,
                      "borderStyle": "solid",
                      "borderWidth": 1,
                      "marginLeft": 8,
                      "minWidth": 64,
                    }
                  }
                >
                  <View
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": false,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      [
                        {
                          "overflow": "hidden",
                        },
                        false,
                        {
                          "borderRadius": 20,
                        },
                      ]
                    }
                    testID="disconnectButton"
                  >
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "flexDirection": "row",
                            "justifyContent": "center",
                          },
                          undefined,
                        ]
                      }
                    >
                      <Text
                        numberOfLines={1}
                        selectable={false}
                        style={
                          [
                            {
                              "color": "rgba(28, 27, 31, 1)",
                              "fontFamily": "System",
                              "fontSize": 14,
                              "fontWeight": "500",
                              "letterSpacing": 0.1,
                              "lineHeight": 20,
                            },
                            {
                              "textAlign": "left",
                            },
                            {
                              "writingDirection": "ltr",
                            },
                            [
                              {
                                "marginHorizontal": 16,
                                "marginVertical": 9,
                                "textAlign": "center",
                              },
                              false,
                              {
                                "marginHorizontal": 24,
                                "marginVertical": 10,
                              },
                              false,
                              false,
                              {
                                "color": "rgba(103, 80, 164, 1)",
                                "fontFamily": "System",
                                "fontSize": 14,
                                "fontWeight": "500",
                                "letterSpacing": 0.1,
                                "lineHeight": 20,
                              },
                              undefined,
                            ],
                          ]
                        }
                      >
                        Disconnect
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
